docs/README.md

```diff
@@ -0,0 +1,327 @@
+- [StringManifesto.md](/docs/StringManifesto.md):
+  Provides a long-term vision for the `String` type.
+
+### Proposals
+
+<!-- TODO: Is it worth listing these out here? -->
+Old proposals are present in the [/docs/proposals](/docs/proposals) directory.
+More recent proposals are located in the
+[swift-evolution][] repository.
+
+[swift-evolution]: https://github.com/apple/swift-evolution
+
+### Surveys
+
+- [ErrorHandlingRationale.rst](/docs/ErrorHandlingRationale.rst):
+  Surveys error-handling in a variety of languages,
+  and describes the rationale behind the design of error handling in Swift.
+- [weak.rst](/docs/weak.rst):
+  Discusses weak references, including the designs in different languages,
+  and proposes changes to Swift (pre-1.0).
+  <!-- NOTE: Outdated -->
+
+### Archive
+
+These documents are known to be out-of-date and are superseded by other
+documentation, primarily [The Swift Programming Language][].
+They are preserved mostly for historical interest.
+
+- [AccessControl.rst](/docs/AccessControl.swift)
+- [Arrays.rst](/docs/Arrays.rst)
+  <!-- Has additional notes on bridging that may be of general interest? -->
+- [Generics.rst](/docs/Generics.rst)
+- [ErrorHandling.rst](/docs/ErrorHandling.rst)
+- [StringDesign.rst](/docs/StringDesign.rst)
+- [TextFormatting.rst](/docs/TextFormatting.rst)
+
+[The Swift Programming Language]: https://docs.swift.org/swift-book
+
+## External Resources
+
+The official [Swift blog](https://swift.org/blog/) contains a lot of useful information,
+such as how library evolution works
+and how the compiler's new diagnostic architecture is structured,
+helping us provide more precise diagnostics.
+
+TODO: Add a new document ExternalResources.md.
+
+## Uncategorized
+
+### Needs refactoring
+
+The documents in this section might be worth breaking up
+into several documents, and linking one document from the other.
+Breaking up into components will provide greater clarity
+to contributors wanting to add new documentation.
+
+- [ARCOptimization.rst](/docs/ARCOptimization.rst):
+  Covers how ARC optimization works, with several examples.
+  TODO: Not clear if this is intended to be a topic guide
+  or a reference guide.
+- [CompilerPerformance.md](/docs/CompilerPerformance.md):
+  Thoroughly discusses different ways of measuring compiler performance
+  and common pitfalls.
+  TODO: Consider breaking up into one high-level
+  topic guide explaining key concepts and individual how-to guides that
+  can be expanded independently.
+  Also, it's not the right place to explain details of different compiler modes.
+- [DevelopmentTips.md](/docs/DevelopmentTips.md):
+  Contains an assortment of tips for better productivity when working on the
+  compiler.
+  TODO: Might be worthwhile to make a dedicated how-to guide or topic guide.
+  It might also be valuable to introduce the tips in context, and have the
+  topic guide link to all the different tips.
```

Comment:
Seems like, for now, this could fit well under the How-To section? It could even be expanded to be "Tips and How-To Guides"...

Reply:
Not part of this PR but I am working on a "How to get started" guide, based on Hassan's PR and our existing README, which will incorporate some aspects of the tips. You can see an incomplete prototype here: https://github.com/varungandhi-apple/swift/blob/vg-doc-wars/docs/HowToGuides/GettingStarted.md (it still has the out-dated Ubuntu 14.04 stuff; ignore that).

So I think it will address the concern:

> all the stuff related to making one's first contribution were all in a section at the top.

without having to add more things to this section's title.

Reply:
That document is looking _excellent_!

Reply:
Thanks! I'm going to mark this as resolved for now.